raft/src/test/java/org/apache/kafka/raft/KafkaRaftClientTest.java

```diff
@@ -95,14 +98,22 @@ private InetSocketAddress mockAddress(int id) {
     }
 
     private KafkaRaftClient buildClient(Set<Integer> voters) throws IOException {
+        return buildClient(voters, new MockStateMachine());
+    }
+
+    private KafkaRaftClient buildClient(Set<Integer> voters, ReplicatedStateMachine stateMachine) throws IOException {
```

Comment:
Why do we need a helper to pass in the state machine? I think we could always clear the old state machine and reuse it.